defaults/main.yml

```diff
@@ -4,7 +4,7 @@
 # This file also serves as a documentation for such a variables.
 
 # Examples of role input variables:
-postgresql_version: 13
+postgresql_version: "13"
```

Comment:
You should also "cast" `posgresql_version` to `string` anywhere you are using the variable in a comparison.  The reason is that there is no way to force the user to provide a string value, so they may provide a numeric value.  Therefore, you should use the `string` filter to "cast" the value to a string.  For example, if the user sets `postgresql_version: 13` in their inventory, then this comparison will always fail:
```
postgresql_version not in __postgresql_versions_el8
```
so you have to do this instead
```
postgresql_version | string not in __postgresql_versions_el8
```

Same with these locations:
```
tasks/main.yml:18:    - postgresql_version not in __postgresql_versions_el8
tasks/main.yml:26:    - postgresql_version not in __postgresql_versions_el9
tasks/main.yml:36:      regex_search('([0-9]*).', '\\1') | first !=  postgresql_version
```
The reason why usage like `__postgresql_packages: ["@postgresql:{{ postgresql_version }}/server"]` is ok is because the `{{ postgresql_version }}` implicitly does the conversion of the value to string.  It's really only a problem where you are comparing the "raw" value of the variable.

For more information, see https://github.com/redhat-cop/automation-good-practices/tree/main/coding_style
```
Use float, int, and bool filters to "cast" public API variables used in numeric operations in Jinja templates.
```